### PR TITLE
Generalized parsing of FreeText or OntologyTermRef

### PR DESCRIPTION
**Pull request by mkuhring**

### CODE REVIEW

**Comment by holtgrewe** [comment]
I think you should be able to override the namedtuple constructor similar as described here:

- https://stackoverflow.com/a/8322450/84349

However, try to put the `__new__` directly into the class instead of subclassing.

File: altamisa/isatab/models.py [hunk 1]
```diff
@@ -10,6 +10,8 @@
 from datetime import date
 from pathlib import Path
 from typing import Dict, Tuple, NamedTuple, Union
+import collections
```
Comment (holtgrewe) [inline altamisa/isatab/models.py]: By convention, I always order imports:

- standard library [sorted alphabetically by library name]
- empty line
- 3rd party [sorted alphabetically by library name]
- empty line
- project/package-internal imports  [sorted alphabetically by library name]

**Comment by mkuhring** [comment]
Thanks for tip! Overwriting the constructor of collections.namedtuple is possible in contrast to typing.NamedTuple.
Not sure though, if it is a good design choice to include all the checks in the constructor like I did now. Though technically possible, what do you think of returning other types (str, None) in case the object (OntologyTermRef) can/should not be created since values are missing?

**Comment by holtgrewe** [comment]
> Not sure though, if it is a good design choice to include all the checks in the constructor like I did now.

Generally, it is. You can split out the checking code into class functions if you prefer.

> Though technically possible, what do you think of returning other types (str, None) in case the object (OntologyTermRef) can/should not be created since values are missing?

Yes, I think we should have one of (`str`, `NoneType`, `OntologyTermRef`).

**Comment by holtgrewe** [comment]
>> Not sure though, if it is a good design choice to include all the checks in the constructor like I did now.
> Generally, it is. You can split out the checking code into class functions if you prefer.

Another possibility is to create static factory functions. But I prefer to fall back to these if there are fundamentally different constructors, i.e., if you want to construct from file object vs. path to file or when constructing from a metric vs. imperial unit value. This way, the code is more readable as the type of construction is made more explicit.

**Pull request merged**